docs/tutorial/kymotracking.rst

```diff
@@ -0,0 +1,157 @@
+It starts by performing peak detection, performing a grey dilation on the image, and detection which pixels remain
+unchanged. Peaks that fall below a certain intensity threshold are discarded. Since this peak detection operates at a
+pixel granularity, it is followed up by a refinement step to attain subpixel accuracy. This refinement is performed by
+computing an offset from the brightness weighted centroid in a small neighborhood `w` around the pixel.
+
+.. math::
+
+    offset = \frac{1}{m} \sum_{i^2 < w^2} i I(x + i)
+
+Where m is given by:
+
+.. math::
+
+    m = \sum_{i^2 < w^2} I(x + i)
+
+After peak detection the feature points are linked together using a forward search analogous to
+:cite:`mangeol2016kymographclear`. This in contrast with the linking algorithm in :cite:`sbalzarini2005feature`
+which uses a graph based optimization approach. This linking step traverses the kymograph, tracing particles starting
+from each frame.
+
+- The algorithm starts at time frame one (the first pixel column).
+
+- It selects the peak with the highest pixel intensity and initiates the first trace.
+
+- Next, it evaluates the subsequent frame, and computes a connection score for each peak in the next frame (to be specified in more detail later).
+
+- If a peak is found with an acceptable score, the peak is added to the trace.
+
+- When no more candidates are available we look in the next `window` frames to see if we can find an acceptable peak there, following the same procedure.
+
+- Once no more candidates are found in the next `window` frames, the trace is terminated and we proceed by initiating a new trace from the peak which is now the highest.
+
+- Once there are no more peaks in the frame from which we are currently initiating traces, we start initiating traces from the next frame. This process is continued until there are no more peaks left to trace.
+
+The score function is based on a prediction of where we expect future peaks. Based on the peak location of the tip of
+the trace `x` and a velocity `v`, it computes a predicted position over time. The score function assumes a Gaussian
+uncertainty around that prediction, placing the mean of that uncertainty on the predicted extrapolation. The width of
+this uncertainty is given by a base width (provided as sigma) and a growing uncertainty over time given by a diffusion
+rate. This results in the following model for the connection score.
+
+.. math::
+
+    S(x, t) = N\left(x + v t, \sigma_{base} + \sigma_{diffusion} \sqrt{t}\right).
+
+Here `N` refers to a normal distribution. In addition to the model, we also have to set a cutoff, after which we deem
+peaks to be so unlikely to be connected that they shouldn't be. By default, this cutoff is set at two sigma. Scores
+outside this cutoff are set to zero which means they will not be accepted as a new point.
+
+
+Using the greedy algorithm
+--------------------------
```

Comment:
The algorithm explanation is excellent. I enjoyed reading it and it's great to have it layed out here for users. But I would flip the order of 1 Details 2 Usage to 1 Usage 2 Details. Most people reading this will come here primarily for the usage explanation and most of the time people don't need to understand how something works, just how to use it. So let's prioritize the usage explanation on top.

Reply:
Agreed!